libcaf_io/src/io/network/stream.cpp

```diff
@@ -143,6 +144,13 @@ bool stream::handle_read_result(rw_state read_result, size_t rb) {
     case rw_state::indeterminate:
       return false;
     case rw_state::success:
+      // if it is the first rw_state::success after SSL_connect,
+      // the ssl connection is established. We recover previous pending write OP.
+      if(wr_op_backoff_) {
+        backend().add(operation::write, fd(), this);
+        wr_op_backoff_ = false;
+      }
+    case rw_state::ssl_error_want_read:
```

Comment:
Logically, I think `read_some` cannot report `ssl_error_want_read`. That would strike me as an error.

Reply:
Nevermind, read can return this error: https://www.openssl.org/docs/man1.1.1/man3/SSL_read.html.